lib/asciidoctor/pdf/formatted_text/transform.rb

```diff
@@ -7,7 +7,7 @@ class Transform
         LF = ?\n
         ZeroWidthSpace = ?\u200b
         CharEntityTable = { amp: '&', apos: ?', gt: '>', lt: '<', nbsp: ?\u00a0, quot: '"' }
-        CharRefRx = /&(?:(#{CharEntityTable.keys.join '|'})|#(?:(\d\d\d{0,4})|x([a-f\d][a-f\d][a-f\d]{0,3})));/
+        CharRefRx = /&(?:(#{CharEntityTable.keys.join '|'})|#(?:(\d{2,6})|x(\h{2,5})));/
```

Comment:
I wrote it as `\d\d\d{0,4}` because it's faster. Please do the same with `\h` (which I'm glad you added!)

Reply:
Found other places that match hex digits and regularized to this form also.
The speed comment piqued my curiosity and ended with writing up https://agnd.net/blog/ruby-regex-count-benchmark/

Reply:
Wow, I'm really impressed that you did that analysis! That's super awesome. It truly is surprising that unrolling the ranges speeds things up, but observation proves it to be true with this regexp engine.